domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/DateTimeEntryValidatorImpl.java

```diff
@@ -26,7 +29,8 @@ public DisplayContextParameterType getType() {
     }
 
     @Override
-    public void validate(final String parameterValue) throws InvalidDateTimeFormatException {
-        dateTimeFormatParser.parseDateTimeFormat(parameterValue, ALLOWED_CHARACTERS_PATTERN);
+    public void validate(final String parameterValue, String fieldType) throws InvalidDateTimeFormatException {
+        Pattern pattern = ((fieldType.equals(BASE_DATE)) ? NOT_ALLOWED_CHARACTERS_PATTERN_DATE : NOT_ALLOWED_CHARACTERS_PATTERN_DATETIME);
```

Comment:
Some people prefer visitor pattern like this on the interface. 
default void validate(final String parameterValue, String fieldType, DateTimeFormatParser dateTimeFormatParser) { .. } . DateTimeFormatParser logic can be moved inside here as well.
Use it if you prefer :)